### ISSUE TITLE
New hashable instructions

### ISSUE THREAD

**Issue by anahitH** [tum-i22/sip-oblivious-hashing#68]
Improve instruction coverage by hashing more instructions. The only requirement from hash function is that the value being hashed should be of integer type. Extend current logic of OH to hash any instruction/operand of instruction which is integer. 
Make sure there is no redundant hashing. E.g. an instruction is used as an argument in multiple instructions, hence is hashed multiple times.

**Comment by anahitH** [comment]
Updated numbers with new hashable instructions are:
tetris - from 26 oh protected to 39, from 30 short range protected to 39
snake - from 28 to 41 short range oh protected instructions, no increase in global oh protected instructions' number
2048 - no increase in global oh numbers, short range oh protected instructions from 70 to 86

**Comment by anahitH** [comment]
It can be.
This error message is misleading.  This means that data independent instruction we are trying to hash is not of integer or floating type, which in principle is not an error. Probably it will be better to change it from ERR to Warning.

**Comment by mr-ma** [comment]
I believe it was meant to have a safe type check before reaching insertHashBuilder. If we cannot handle an arbitrary data type why bother insertHashBuilder?

**Comment by anahitH** [comment]
Agree, it will be more clear if this check is done before calling insertHashBuilder. 
Anyway this has nothing to do with this issue, right?

**Comment by mr-ma** [comment]
I thought maybe the commit for new hashable instructions (this issue) is causing this error.

**Comment by anahitH** [comment]
I've modified instruction OH instrumentation process to first check the validity of instruction type being hashed. If the type is not integer or float instruction is recorded as "non-hashable" and insertHashBuilder is not called. The error message mentioned above is removed. Instead there is a info/warning message.

**Comment by mr-ma** [comment]
<a href="https://github.com/anahitH"><img src="https://avatars1.githubusercontent.com/u/23015948?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [anahitH](https://github.com/anahitH)**
_Friday May 11, 2018 at 23:15 GMT_

----

Updated numbers with new hashable instructions are:
tetris - from 26 oh protected to 39, from 30 short range protected to 39
snake - from 28 to 41 short range oh protected instructions, no increase in global oh protected instructions' number
2048 - no increase in global oh numbers, short range oh protected instructions from 70 to 86

**Comment by mr-ma** [comment]
<a href="https://github.com/mr-ma"><img src="https://avatars1.githubusercontent.com/u/4548085?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mr-ma](https://github.com/mr-ma)**
_Tuesday May 29, 2018 at 14:42 GMT_

----

@anahitH  ERR. Any value other than int and float is passed to insertHashBuilder
Could this be related to your changes?

**Comment by mr-ma** [comment]
<a href="https://github.com/anahitH"><img src="https://avatars1.githubusercontent.com/u/23015948?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [anahitH](https://github.com/anahitH)**
_Tuesday May 29, 2018 at 14:48 GMT_

----

It can be.
This error message is misleading.  This means that data independent instruction we are trying to hash is not of integer or floating type, which in principle is not an error. Probably it will be better to change it from ERR to Warning.

**Comment by mr-ma** [comment]
<a href="https://github.com/mr-ma"><img src="https://avatars1.githubusercontent.com/u/4548085?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mr-ma](https://github.com/mr-ma)**
_Tuesday May 29, 2018 at 15:08 GMT_

----

I believe it was meant to have a safe type check before reaching insertHashBuilder. If we cannot handle an arbitrary data type why bother insertHashBuilder?

**Comment by mr-ma** [comment]
<a href="https://github.com/anahitH"><img src="https://avatars1.githubusercontent.com/u/23015948?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [anahitH](https://github.com/anahitH)**
_Tuesday May 29, 2018 at 15:21 GMT_

----

Agree, it will be more clear if this check is done before calling insertHashBuilder. 
Anyway this has nothing to do with this issue, right?

**Comment by mr-ma** [comment]
<a href="https://github.com/mr-ma"><img src="https://avatars1.githubusercontent.com/u/4548085?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [mr-ma](https://github.com/mr-ma)**
_Tuesday May 29, 2018 at 16:04 GMT_

----

I thought maybe the commit for new hashable instructions (this issue) is causing this error.

**Comment by mr-ma** [comment]
<a href="https://github.com/anahitH"><img src="https://avatars1.githubusercontent.com/u/23015948?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [anahitH](https://github.com/anahitH)**
_Tuesday May 29, 2018 at 18:53 GMT_

----

I've modified instruction OH instrumentation process to first check the validity of instruction type being hashed. If the type is not integer or float instruction is recorded as "non-hashable" and insertHashBuilder is not called. The error message mentioned above is removed. Instead there is a info/warning message.
